test-network-function/networking/suite_test.go

```diff
@@ -0,0 +1,161 @@
+		},
+	}
+
+	for _, tc := range testCases {
+		err := parseVariables(tc.inputRes, tc.declaredPorts)
+		assert.Nil(t, err)
+		assert.Equal(t, tc.expectedDeclaredPorts, tc.declaredPorts)
+	}
+}
+
+func TestDeclaredPortList(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputs
+		container     int
+		podName       string
+		podNamespace  string
+		declaredPorts map[key]string
+
+		// expected outputs here
+		expectedDeclaredPorts map[key]string
+	}{
+		{
+			container:             0,
+			podName:               "test-54bc4c6d7-2gwlz",
+			podNamespace:          "tnf",
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{{port: 8080, protocol: "TCP"}: "http-probe"},
+		},
+		{
+			container:             0,
+			podName:               "",
+			podNamespace:          "",
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{},
+		},
+	}
+
+	for _, tc := range testCases {
+		err := declaredPortList(tc.container, tc.podName, tc.podNamespace, tc.declaredPorts)
+		assert.Nil(t, err)
+		assert.Equal(t, tc.expectedDeclaredPorts, tc.declaredPorts)
+	}
+}
+
+func TestListeningPortList(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputs
+		commandlisten  []string
+		nodeOc         *interactive.Context
+		listeningPorts map[key]string
+
+		// expected outputs here
+		expectedlisteningPorts map[key]string
+	}{
+		{
+			commandlisten:          []string{"nsenter -t 4380 -n", "ss -tulwnH"},
+			nodeOc:                 nil,
+			listeningPorts:         map[key]string{},
+			expectedlisteningPorts: map[key]string{{port: 8080, protocol: "TCP"}: ""},
+		},
+		{
+			commandlisten:          []string{},
+			nodeOc:                 nil,
+			listeningPorts:         map[key]string{},
+			expectedlisteningPorts: map[key]string{},
+		},
+	}
+	for _, tc := range testCases {
+		err := listeningPortList(tc.commandlisten, tc.nodeOc, tc.listeningPorts)
+		assert.Nil(t, err)
+		assert.Equal(t, tc.expectedlisteningPorts, tc.listeningPorts)
+	}
+}
+
+func TestCheckIfListenIsDeclared(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputs
+		listeningPorts map[key]string
+		declaredPorts  map[key]string
+
+		// expected outputs here
+		expectedres map[key]string
+	}{
+		{
+			listeningPorts: map[key]string{},
+			declaredPorts:  map[key]string{},
+			expectedres:    map[key]string{},
+		},
+		{
+			listeningPorts: map[key]string{{port: 8080, protocol: "TCP"}: ""},
+			declaredPorts:  map[key]string{{port: 8080, protocol: "TCP"}: "http-probe"},
+			expectedres:    map[key]string{},
+		},
+		{
+			listeningPorts: map[key]string{{port: 8080, protocol: "TCP"}: ""},
+			declaredPorts:  map[key]string{},
+			expectedres:    map[key]string{{port: 8080, protocol: "TCP"}: ""},
+		},
+	}
+	for _, tc := range testCases {
+		err := checkIfListenIsDeclared(tc.listeningPorts, tc.declaredPorts)
```

Comment:
The checkIfListenIsDeclared() function does never return an error. It's always returning an empty map, which will never be nil. That also makes the assertion on line 158 useless, since it will never fail. I suggest changing this to:
```
listeningPortsNotDeclared := checkIfListenIsDeclared(tc.listeningPorts, tc.declaredPorts)
```